app/domain/search/audit_filter.rb

```diff
@@ -0,0 +1,42 @@
+class Search
+  class AuditFilter
+    AUDIT_STATUSES_IDENTIFIERS = [
+      :audited,
+      :non_audited,
+    ].freeze
+
+    attr_accessor :status
+
+
+    def initialize(status)
+      self.status = status ? status.to_sym : nil
+
+      raise_if_unrecognised_audit_status
+    end
+
+    def apply(scope)
+      if self.status == :audited
+        scope.joins(:audit)
+      elsif self.status == :non_audited
+        scope.left_outer_joins(:audit).where("content_items.content_id not IN( select content_id from audits)")
+      else
+        scope
+      end
+    end
+
+    def raise_if_unrecognised_audit_status
+      if status && !AUDIT_STATUSES_IDENTIFIERS.include?(status)
+        raise AuditStatusError, "unrecognised audit status: #{status}"
+      end
+    end
+
+    def self.all_status
+      [
+        OpenStruct.new(id: 'audited', name: "Audited"),
+        OpenStruct.new(id: 'non_audited', name: "Non Audited"),
```

Comment:
Please can we avoid `OpenStruct`? I don't like that a simple typo can make its way all the way into production before someone realises there's a problem, e.g. `OpenStruct.new(foo: "foo").fooo => nil`. I think it's better if a `NoMethodError` is raise much earlier during development.

Perhaps we could pair on this?